app/scripts/app-init.js

```diff
@@ -111,14 +110,20 @@ function importAllScripts() {
   }
 }
 
+// Ref: https://stackoverflow.com/questions/66406672/chrome-extension-mv3-modularize-service-worker-js-file
 // eslint-disable-next-line no-undef
 self.addEventListener('install', importAllScripts);
```

Comment:
I am wondering why we use `self.addEventListener('install', ...` here instead of `chrome.runtime.onInstalled.addListener`, and `chrome.runtime.onMessage.addListener` instead of  `self.addEventListener('message', ...`